File: solutions/go/parsing-log-files/3/parsing_log_files.go
package parsinglogfiles

import (
    "regexp"
    "fmt"
)

var RegexIsValidLine = regexp.MustCompile(`^\[TRC\]|^\[DBG\]|^\[INF\]|^\[WRN\]|^\[ERR\]|^\[FTL\]`)
var RegexSplitLogLine = regexp.MustCompile(`<[~*=-]*>`)
var RegexCountQuotedPasswords = regexp.MustCompile(`(?i)".*password.*"`)
var RegexRemoveEndOfLineText = regexp.MustCompile(`end-of-line[\d]+`)
var RegexTagWithUserName = regexp.MustCompile(`User[\s]+(\S*)\b`)

func IsValidLine(text string) bool {
    return RegexIsValidLine.MatchString(text)
}

func SplitLogLine(text string) []string {
    return RegexSplitLogLine.Split(text, -1)
}

func CountQuotedPasswords(lines []string) int {
    cnt := 0
    for _, v := range lines {
        if RegexCountQuotedPasswords.MatchString(v) {
            cnt++
        }
    }
    return cnt
}

func RemoveEndOfLineText(text string) string {
    return RegexRemoveEndOfLineText.ReplaceAllString(text, "")
}

func TagWithUserName(lines []string) []string {
    res := make([]string, 0)
    for _, v := range lines {
        mat := RegexTagWithUserName.FindStringSubmatch(v)
        if len(mat) > 1 {
            res = append(res, fmt.Sprintf("[USR] %v %v", mat[1], v))
        } else {
            res = append(res, v)
        }
    }
    return res
}

File: solutions/csharp/parsing-log-files/3/ParsingLogFiles.cs
using System.Text.RegularExpressions;

public class LogParser
{
    private readonly Regex validLine = new(@"^\[(TRC|DBG|INF|WRN|ERR|FTL)\]", RegexOptions.Compiled);
    public bool IsValidLine(string text) => validLine.IsMatch(text);

    private readonly Regex delimiter = new(@"<[\^*=-]+>", RegexOptions.Compiled);
    public string[] SplitLogLine(string text) => delimiter.Split(text);

    private readonly Regex password = new(@""".*password.*""", RegexOptions.IgnoreCase | RegexOptions.Compiled);
    public int CountQuotedPasswords(string lines) => lines.Split(Environment.NewLine)
        .Count(line => password.IsMatch(line));

    private readonly Regex eol = new(@"end-of-line[0-9]+", RegexOptions.Compiled);
    public string RemoveEndOfLineText(string line) => eol.Replace(line, "");

    private readonly Regex offending = new(@"\bpassword\S+", RegexOptions.IgnoreCase | RegexOptions.Compiled);
    public string[] ListLinesWithPasswords(string[] lines) => lines.Select(line =>
        {
            Match match = offending.Match(line);
            return match.Success ? $"{match.Value}: {line}" : $"--------: {line}";
        }).ToArray();
}